system/lib/pthread/library_pthread.c

```diff
@@ -898,6 +893,15 @@ EM_JS(void, initPthreadsJS, (void), {
 EMSCRIPTEN_KEEPALIVE
 __attribute__((constructor(48)))
 void __emscripten_pthread_data_constructor(void) {
-  initPthreadsJS();
-  pthread_self()->locale = &libc.global_locale;
+  initPthreadsJS(&__main_pthread);
+  // The pthread struct has a field that points to itself - this is used as
+  // a magic ID to detect whether the pthread_t structure is 'alive'.
+  __main_pthread.self = &__main_pthread;
+      // pthread struct robust_list head should point to itself.
+  __main_pthread.robust_list.head = &__main_pthread.robust_list;
```

Comment:
```suggestion
  __main_pthread.robust_list.head = &__main_pthread.robust_list.head;
```
This would avoid depending on `head` being the initial field, or is that bad for some reason?

Reply:
Honestly I was just trying to mimic exactly what the JS code was doing.  I don't really understand the purpose of this assignment.

Reply:
Me either, but the actual address is identical in both cases - just that writing `.head` would still work if we reorder the struct in the future?